src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImpl.java

```diff
@@ -188,18 +190,26 @@ private void extractSchema(
 
   private ImmutableCollection<String> getScriptNames(Arguments arguments) {
     ImmutableSet<String> allScriptNames = scriptManager.getAllScriptNames();
+    ImmutableList<String> requiredScriptNames =
+        allScriptNames.stream()
+            .filter(name -> !name.startsWith(OPTIONAL_SCRIPT_PREFIX))
+            .collect(toImmutableList());
+    ImmutableSet<String> scriptNamesToProcess = arguments.includeOptionalScripts()
+        ? allScriptNames
+        : ImmutableSet.copyOf(requiredScriptNames);
+
     if (arguments.sqlScripts().isEmpty()) {
       if (arguments.skipSqlScripts().isEmpty()) {
-        return allScriptNames;
+        return scriptNamesToProcess;
       } else {
         ImmutableList<String> skip =
-            validateScriptNames("skip-sql-scripts", allScriptNames, arguments.skipSqlScripts());
-        return allScriptNames.stream()
+            validateScriptNames("skip-sql-scripts", scriptNamesToProcess, arguments.skipSqlScripts());
+        return scriptNamesToProcess.stream()
             .filter(name -> !skip.contains(name))
             .collect(toImmutableList());
```

Comment:
Consider simplifying to `Sets.difference(scriptNamesToProcess, skip)` instead of `stream().filter`.